unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/BaseAvatar.cs

```diff
@@ -77,16 +77,17 @@ UniTask GetRevealTask(Material material, float revealPosition, float completionP
                                .ToUniTaskInstantCancelation(true, cancellationToken: linkedCts.Token);
             }
 
-            baseAvatarReferences.ParticlesContainer.SetActive(true);
-            List<UniTask> tasks = new List<UniTask>();
-            tasks.Add(GetRevealTask(ghostMaterial, avatarHeight, completionHeight));
-
-            for (var index = 0; index < cachedMaterials.Count; index++)
+            try
             {
-                tasks.Add(GetRevealTask(cachedMaterials[index], -avatarHeight, -completionHeight));
-            }
+                baseAvatarReferences.ParticlesContainer.SetActive(true);
+                List<UniTask> tasks = new List<UniTask>();
```

Comment:
let's fill it on creation
```
List<UniTask> tasks = new List<UniTask>
    { GetRevealTask(ghostMaterial, avatarHeight, completionHeight) };
```

Reply:
I like it explicit there because it works as an analogy for the different tasks (we have to apply to the ghostMaterial) te same thing we do to the mesh materials.